pkg/networkpolicy/networkpolicy_test.go

```diff
@@ -0,0 +1,37 @@
+package networkpolicy
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+)
+
+func TestLoad(t *testing.T) {
+	match, err := filepath.Glob("testdata/*.input")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, inputFile := range match {
+
+		a := NewAdvisor()
+
+		err := a.LoadFile(inputFile)
+		if err != nil {
+			t.Fatal(err)
+		}
+		a.GeneratePolicies()
+		generatedOuput := a.FormatPolicies()
+
+		goldenFile := inputFile[:len(inputFile)-len(".input")] + ".golden"
+		goldenOutputBytes, err := ioutil.ReadFile(goldenFile)
+		if err != nil {
+			t.Fatal(err)
+		}
+		goldenOutput := string(goldenOutputBytes)
+
+		if generatedOuput != goldenOutput {
```

Comment:
I'm a little bit worried about this approach, I don't think comparing strings is that robust, a very simple change on the way marshall works could break it.

I tried to unmarshall to `networkingv1.NetworkPolicy` and then compare with `reflect.DeepEqual`, but somehow they are not equal (one of them as nil while other has empty maps / slices.

Reply:
If the way marshall work changes, we would notice it because the test would fail, and we can just update the test.

At some point, I was pondering about outputting yaml with comments, to advise the user to create the label in the namespace, for cross namespace communication. In that case, string comparison would test that too.

Reply:
The marshalling seems stable: keys seems to be sorted in alphabetical order.